app/src/main/kotlin/com/wire/android/ui/authentication/create/code/CodeScreen.kt

```diff
@@ -0,0 +1,113 @@
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxHeight
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.input.TextFieldValue
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.text.style.TextDecoration
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.wire.android.R
+import com.wire.android.ui.authentication.create.code.CodeViewModel
+import com.wire.android.ui.authentication.create.code.CodeViewState
+import com.wire.android.ui.common.textfield.CodeTextField
+import com.wire.android.ui.common.textfield.WireTextFieldState
+import com.wire.android.ui.common.topappbar.WireCenterAlignedTopAppBar
+import com.wire.android.ui.theme.wireDimensions
+import com.wire.android.ui.theme.wireTypography
+
+@Composable
+fun CodeScreen(viewModel: CodeViewModel, title: String) {
+    CodeContent(
+        state = viewModel.codeState,
+        title = title,
+        onCodeChange = viewModel::onCodeChange,
+        onResendCodePressed = viewModel::onResendCodePressed,
+        onBackPressed = viewModel::goBackToPreviousStep,
+    )
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+private fun CodeContent(
+    state: CodeViewState,
+    title: String,
+    onCodeChange: (TextFieldValue) -> Unit,
+    onResendCodePressed: () -> Unit,
+    onBackPressed: () -> Unit,
+) {
+    Scaffold(topBar = { WireCenterAlignedTopAppBar(elevation = 0.dp, title = title, onNavigationPressed = onBackPressed) },) {
+        Column(
+            horizontalAlignment = Alignment.CenterHorizontally,
+            verticalArrangement = Arrangement.Top,
+            modifier = Modifier.fillMaxHeight()
+        ) {
+            Text(
+                text = stringResource(R.string.create_personal_account_code_text, state.email),
+                style = MaterialTheme.wireTypography.body01,
+                modifier = Modifier.fillMaxWidth().padding(
+                        horizontal = MaterialTheme.wireDimensions.spacing16x,
+                        vertical = MaterialTheme.wireDimensions.spacing24x
+                    )
+            )
+            Spacer(modifier = Modifier.weight(1f))
+            Column(horizontalAlignment = Alignment.CenterHorizontally,) {
+                CodeTextField(
+                    codeLength = CodeViewModel.CODE_LENGTH,
```

Comment:
I am not sure if such information should belong to the viewmodel L ?

Reply:
ViewModel also needs to know if the code is fully filled, but I managed to change it a bit - now `onCodeChange` returns value and flag indicating if it's filled